client/src/components/DisplayTable/DisplayTable.tsx

```diff
@@ -96,13 +103,40 @@ type ColumnConfig = Partial<ReturnType<typeof get_column_config_defaults>> & {
   formatter: FormatKey | ((val: CellValue) => React.ReactNode);
 };
 
+const get_default_dropdown_filter = _.memoize(
+  (col_configs_with_defaults: ColumnConfigs, data: DisplayTableData[]) =>
```

Comment:
Oh hmmm, wouldn't having `data` as first arg make caching sufficient? So that on the same set of data (which would mean on the same table), this would always be cached. When the table data changes, cache will change so it'll be fresh default dropdown filter with the new data.

Reply:
Doesn't feel like a safe assumption to make. It's possible that either/or of the configs or data could be updated from outside the table (if not currently, then still possible in the future). Not doing this carefully right now could be a future (and hard to spot) bug.

Reply:
How would I go about writing the resolver? I can only think of stringify'ing everything and concatenating them but that can't be the most efficient way

Reply:
That's most robust, but if you don't mind the extra maintenance burden then you can make it smarter by just keying on characteristics that you know change the output of to the memoized function (may or may not be more efficient, mostly better if different inputs can result in the same output when similar n known ways). Optimization often means sacrificing ease of maintenance/doing something hacky :shrug:. That's at least a localized problem, better than writing it in a way that could produce bugs if assumptions about other code are broken.

Is the memoized function particularly slow? Could it be optimized in any way?